lib/engine/game/g_1847_ae/round/draft.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_relative '../../../round/draft'
+
+module Engine
+  module Game
+    module G1847AE
+      module Round
+        class Draft < Engine::Round::Draft
+          def initialize(game, steps, **opts)
+            super
+
+            @players_reordered_for_next_round = false
+          end
+
+          def next_entity_index!
+            # First round of draft is perforemd in reverse player order, then it must be reversed back to normal
+            if @entity_index == @entities.size - 1 && !@game.draft_first_round_finished
+              @entities.reverse!
+              @game.draft_first_round_finished = true
+            end
+
+            super
+          end
+
+          def process_action(action, suppress_log = false)
+            type = action.type
+            clear_cache!
+
+            before_process(action)
+
+            step = @steps.find do |s|
+              next unless s.active?
+
+              process = s.actions(action.entity).include?(type)
+              blocking = s.blocking?
+
+              raise GameError, "Blocking step #{s.description} cannot process action #{action.id}" if blocking && !process
+
+              blocking || process
+            end
+
+            raise GameError, "No step found for action #{type} at #{action.id}: #{action.to_h}" unless step
+
+            step.acted = true
+            step.send("process_#{action.type}", action, suppress_log)
+
+            @at_start = false
+
+            after_process_before_skip(action)
+            skip_steps
+            clear_cache!
+            after_process(action)
+          end
+
+          def reset_entity_index!
+            @game.next_turn!
+            @entity_index = (@entity_index + 1) % @entities.size
+
+            return if @players_reordered_for_next_round
+
+            # Reorder players for the next round of draft
+            @game.reorder_players
+            @players_reordered_for_next_round = true
```

Comment:
What's the purpose of this `@players_reordered_for_next_round` check? I can't see how if this will ever be true when it is checked, this method should only ever be called once.
It would also be more consistent with the style of other games to have the `reorder_players` call in `next_round!` instead of here.

Reply:
It seems to get called N-1 times where N is the number of players, that's why the check was there. 

I moved the reorder to `next_round!`, good suggestion.